src/snowflake/snowpark/functions.py

```diff
@@ -340,35 +340,36 @@ def column(name1: str, name2: Optional[str] = None, _emit_ast: bool = True) -> C
 
 @publicapi
 def lit(
-    literal: LiteralType, datatype: Optional[DataType] = None, _emit_ast: bool = True
+    literal: ColumnOrLiteral,
```

Comment:
this should be type hint `Any` to be compatible with PySpark.

Reply:
Even though pyspark says it supports any, it raises TypeErrors for unsupported columns. In term of behavior, we are compatible

Reply:
@sfc-gh-aalam can we address this? Why not use `Any` as type hint?

Reply:
Because `ColumnOrLiteral` is the accurate type hint.

Pyspark's ref can be seen here: https://spark.apache.org/docs/3.5.0/api/python/_modules/pyspark/sql/functions.html#lit